### PR TITLE
Feat/scaling mvp

### PR DESCRIPTION
**Pull request by zachferland**
Changes before merging 

- [ ]  change circle CI config branches, and set dev prod env correctly (right now pushes to this branch deploy to test infra)

### CODE REVIEW

**Comment by zachferland** [comment]
@oed @lsenta this is basically all the code for the scaling release, and rebased on develop. As we have discussed, it pretty minimal in the end. No rush at the moment, just pinging to say it is in a state that it can be reviewed anytime now. Didn't do reviews for each story along the way. 

Probably important to also take a look at https://github.com/3box/orbit-db-cache-redis

Looks like we will aim to merge this into develop after the timestamp stories are merged and released.

**Comment by zachferland** [comment]
@oed @lsenta review when you can now, will want to merge in soon. I think right after we get @lsenta stories in for meta data and release, we can merge this in.

File: src/pinning.js [hunk 1]
```diff
@@ -16,19 +17,22 @@ const IPFS_OPTIONS = {
...
+    this.runCacheServiceOnly = runCacheServiceOnly
   }
 
   async start () {
     this.ipfs = await this._initIpfs()
     const ipfsId = await this.ipfs.id()
     console.log(ipfsId)
-    this.orbitdb = new OrbitDB(this.ipfs, this.orbitdbPath)
+    const orbitOpts = this.orbitCacheOpts ? { cache: orbitDBCache(this.orbitCacheOpts) } : { }
+    this.orbitdb = new OrbitDB(this.ipfs, this.orbitdbPath, orbitOpts)
```
Comment (oed) [inline src/pinning.js]: The path is not used in our orbit-db-cache impl right? Makes sense to include it here though if someone where to use a custom cache though.

File: package.json [hunk 2]
```diff
@@ -24,13 +24,16 @@
   "homepage": "https://github.com/uport-project/3box-pinning-server#readme",
   "dependencies": {
     "analytics-node": "^3.3.0",
+    "aws-sdk": "^2.414.0",
     "axios": "^0.18.0",
+    "datastore-s3": "^0.2.4",
     "dotenv": "^6.2.0",
     "exectimer": "^2.2.1",
     "express": "^4.16.4",
     "get-folder-size": "^2.0.0",
```
Comment (oed) [inline package.json]: Can you remove this dep?

File: src/pinning.js (hunk 1 above)
Comment (zachferland) [reply src/pinning.js]: @oed  were you saying pass it through to orbitDBCache? i guess, but there is no config here for generally passing in a cache, and only two caches exist so far, the orbit default and this one. If not orbit redis path is given, then it fallbacks to the default

Comment (oed) [reply src/pinning.js]: nvm this comment, I think I misread the code.

File: package.json (hunk 2 above)
Comment (oed) [reply package.json]: "get-folder-size" that is. Shoudn't be needed anymore.

**Comment by oed** [comment]
Closing in favor of #127

**Pull request closed without merging**